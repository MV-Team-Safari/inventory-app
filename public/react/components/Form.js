import React, { useState } from "react";
import apiURL from "../api";

export const Form = () => {
  const [title, setTitle] = useState("");
  const [description, setDescription] = useState("");
  const [category, setCategory] = useState("");
  const [price, setPrice] = useState("");
  const [imgURL, setImgURL] = useState("");

  async function handleSubmit(e) {
    e.preventDefault();
    const data = { title, description, price, category, imgURL };
    console.log(data);
    const response = await fetch(`${apiURL}/items`, {
      method: "POST",

      headers: {
        "Content-Type": "application/json",
      },

      body: JSON.stringify(data),
    });
    return response.json();
  }

  return (
    <div className="form-container">
      <form className="form" onSubmit={handleSubmit}>
        <label className="form-item-name">
          New Item Name/Title:
          <input
            className="form-item-name-input"
            type="text"
            name="item-name"
            value={title}
            onChange={(event) => setTitle(event.target.value)}
          />
        </label>
        <label className="form-item-desc">
          New Item Description:
          <input
            className="form-item-desc-input"
            type="text"
            name="item-description"
            value={description}
            onChange={(event) => setDescription(event.target.value)}
          />
        </label>
        <label className="form-item-price">
          New Item Price:
          <input
            className="form-item-price-input"
            type="number"
            name="item-price"
            value={price}
            onChange={(event) => setPrice(event.target.value)}
          />
        </label>
        <label className="form-item-category">
          New Item Category:
          <input
            className="form-item-category-input"
            type="text"
            name="item-category"
            value={category}
            onChange={(event) => setCategory(event.target.value)}
          />
        </label>
        <label className="form-item-img">
          Image URL:
          <input
            className="form-item-img-input"
            type="text"
            name="item-img-url"
            value={imgURL}
            onChange={(event) => setImgURL(event.target.value)}
          />
        </label>
        <input
          className="form-item-submit-btn"
          type="submit"
          value="Submit Your Item"
        />
      </form>
    </div>
  );
};
