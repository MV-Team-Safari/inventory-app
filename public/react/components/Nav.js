import React from "react";

export default function Nav() {
  return (
    <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
      <div className="container">
        <button
          className="navbar-view-all-btn"
          type="button"
          data-toggle="collapse"
        >
          View All
        </button>
        <button className="navbar-add-btn" type="button" data-toggle="collapse">
          Add
        </button>
      </div>
      <div className="nav-title">
        <h1>Item Store</h1>
        <h2>All things ðŸ”¥</h2>
      </div>

      <a className="navbar-user" href="/">
        <img
          className="navbar-user-img"
          src="https://fairtradesafaris.com/wp-content/uploads/2021/06/1097003_FTS-JuneBlogs-10trees-Umbrella-Thorn_061521.png"
        />
      </a>
    </nav>
  );
}
