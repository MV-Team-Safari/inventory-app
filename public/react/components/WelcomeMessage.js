import React from "react";

export default function WelcomeMessage() {
  return (
    <>
      <div id="main-area" className="welcome-message">
        <div className="welcome-message-content">
          <h1>Welcome to your inventory!</h1>
          <p>Click View All to view all of your items.</p>
          <p>Click Add to add a new item to your inventory.</p>
          <img src="https://media.tenor.com/J49ROi-SxSMAAAAC/the-office-steve-carell.gif" />
        </div>
      </div>
    </>
  );
}
