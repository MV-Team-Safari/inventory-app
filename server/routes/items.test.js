const { Item } = require("../models")
const { sequelize } = require("../db");
import 'regenerator-runtime/runtime'
import renderer from 'react-test-renderer';
import Nav from "../../public/react/components/Nav";
import React from "react"
import WelcomeMessage from "../../public/react/components/WelcomeMessage";
import Form from "../../public/react/components/Form";

describe("Model Test", () => {

    beforeEach( async () => {
        await sequelize.sync({ force: true });
        await Item.create({
            title: "Test Item",
            description: "An item to test the funcitonality of this App.",
            price: 67.89,
            category: "Project Tests",
            image: "https://fakestoreapi.com/img/71-3HjGNDUL._AC_SY879._SX._UX._SY._UY_.jpg"
        })
    })

    test("Should Create New Entry", async () => {
        const receivedItem = await Item.findByPk(1);
        expect(receivedItem.title).toBe("Test Item");
    })

    test("Should Update New Entry", async () => {
        const receivedItem = await Item.findByPk(1);
        await receivedItem.update({ category: "New Category" });
        const updatedItem = await Item.findByPk(1)
        expect(updatedItem.category).toBe("New Category");
    })

    test("Should Delete New Entry", async () => {
        const receivedItem = await Item.findByPk(1);
        await receivedItem.destroy();
        const foundItem = await Item.findAll();
        expect(foundItem).toEqual([]);
    })

    test("Should Read New Entry", async () => {
        const receivedItem = await Item.findByPk(1);
        expect(receivedItem.title).toBe("Test Item")
    })
})

describe("React Tests", () => {

    test("Should Render Nav Component", () => {
        const component = renderer.create(<Nav
          />)
          console.log(component)
          let tree = component.toJSON();
          expect(tree).toMatchSnapshot()
    })

    test("Should Render WelcomeMessage Component", () => {
        const component = renderer.create(<WelcomeMessage
          />)
          console.log(component)
          let tree = component.toJSON();
          expect(tree).toMatchSnapshot()
    })

    // test("Should Render Form Component", () => {
    //     const component = renderer.create(<Form
    //       />)
    //       console.log(component)
    //       let tree = component.toJSON();
    //       expect(tree).toMatchSnapshot()
    // })
})