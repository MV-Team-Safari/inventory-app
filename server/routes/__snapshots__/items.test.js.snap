// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React Tests Should Render Form Component 1`] = `
<div
  className="form-container"
>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <label
      className="form-item-name"
    >
      New Item Name/Title:
      <input
        className="form-item-name-input"
        name="item-name"
        onChange={[Function]}
        type="text"
        value=""
      />
    </label>
    <label
      className="form-item-desc"
    >
      New Item Description:
      <input
        className="form-item-desc-input"
        name="item-description"
        onChange={[Function]}
        type="text"
        value=""
      />
    </label>
    <label
      className="form-item-price"
    >
      New Item Price:
      <input
        className="form-item-price-input"
        name="item-price"
        onChange={[Function]}
        type="number"
        value=""
      />
    </label>
    <label
      className="form-item-category"
    >
      New Item Category:
      <input
        className="form-item-category-input"
        name="item-category"
        onChange={[Function]}
        type="text"
        value=""
      />
    </label>
    <label
      className="form-item-img"
    >
      Image URL:
      <input
        className="form-item-img-input"
        name="item-img-url"
        onChange={[Function]}
        type="text"
        value=""
      />
    </label>
    <input
      className="form-item-submit-btn"
      type="submit"
      value="Submit New Item"
    />
  </form>
</div>
`;

exports[`React Tests Should Render Nav Component 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-dark bg-dark"
>
  <div
    className="container"
  >
    <button
      className="navbar-view-all-btn"
      data-toggle="collapse"
      onClick={[Function]}
      type="button"
    >
      View All
    </button>
    <button
      className="navbar-add-btn"
      data-toggle="collapse"
      onClick={[Function]}
      type="button"
    >
      Add Item
    </button>
  </div>
  <div
    className="nav-title"
  >
    <h1>
      The Object Shop
    </h1>
    <h2>
      Find your favorite things
    </h2>
  </div>
  <a
    className="navbar-user"
    href="/"
  >
    <img
      className="navbar-user-img"
      src="https://www.repicture.com/assets/images/default.png"
    />
  </a>
</nav>
`;

exports[`React Tests Should Render WelcomeMessage Component 1`] = `
<div
  className="welcome-message"
>
  <div
    className="welcome-message-content"
  >
    <h1
      className="welcome-message-content-heading"
    >
      Welcome To Your Inventory!
    </h1>
    <p
      className="welcome-message-content-p1"
    >
      Click 
      <span
        className="grey"
      >
        View All Button
      </span>
       to view all of your items.
    </p>
    <p
      className="welcome-message-content-p2"
    >
      Click 
      <span
        className="grey"
      >
        Add Item Button
      </span>
       to add a new item to your inventory.
    </p>
    <img
      className="welcome-message-img"
      src="https://media.tenor.com/J49ROi-SxSMAAAAC/the-office-steve-carell.gif"
    />
  </div>
</div>
`;
